### PR TITLE
Add GML and GraphML NetworkX datasets

### PR DESCRIPTION
**Pull request by sbrugman**
## Description
Introduces feature described in https://github.com/quantumblacklabs/kedro/issues/828

## Development notes
Follow up from https://github.com/quantumblacklabs/kedro/pull/830 with comments processed, based on the lastest development branch. Thanks @ignacioparicio for reviewing!
Omitted GPickleDataSet as GPickle is deprecated in NetworkX and will be removed in the third version ([ref](https://networkx.org/documentation/stable/_modules/networkx/readwrite/gpickle.html#write_gpickle)).
Included GraphML in addition to GML. Some tools prefer GraphML as it's able to perfectly preserve internal property maps ([ref](https://graph-tool.skewed.de/static/doc/graph_tool.html#graph_tool.load_graph)).

## Checklist

- [X] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- [X] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [X] Updated the documentation to reflect the code changes
- [X] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- [X] Added tests to cover my changes

## Notice

- [X] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -19,6 +19,8 @@
 | Type                | Description                                                    | Location                       |
 | ------------------- | -------------------------------------------------------------- | ------------------------------ |
 | `pandas.XMLDataSet` | Read XML into Pandas DataFrame. Write Pandas DataFrame to XML. | `kedro.extras.datasets.pandas` |
+| `networkx.GraphMLDataSet`       | Works with networkx graph XML                      | `kedro.extras.datasets.networkx` |
+| `networkx.GMLDataSet`       | Works with networkx graph modelling language           | `kedro.extras.datasets.networkx` |
```
Comment (AntonyMilneQB) [inline RELEASE.md:23]: For consistency with other bits of our RELEASE.md 🙂 
```suggestion
| `networkx.GraphMLDataSet`       | Work with NetworkX using GraphML files                      | `kedro.extras.datasets.networkx` |
| `networkx.GMLDataSet`       | Work with NetworkX using Graph Modelling Language files         | `kedro.extras.datasets.networkx` |
```

File: kedro/extras/datasets/networkx/__init__.py [hunk 2]
```diff
@@ -27,11 +27,17 @@
...
 
 with suppress(ImportError):
-    from .networkx_dataset import NetworkXDataSet  # NOQA
+    from .json_dataset import JSONDataSet  # NOQA
+
+with suppress(ImportError):
+    from .gml_dataset import GMLDataSet  # NOQA
+
+with suppress(ImportError):
+    from .graphml_dataset import GraphMLDataSet  # NOQA
```
Comment (AntonyMilneQB) [inline kedro/extras/datasets/networkx/__init__.py:43]: Please could you update this as per the open PR https://github.com/quantumblacklabs/kedro/pull/874 to remove the `# NOQA`?

File: kedro/extras/datasets/networkx/gml_dataset.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        fs_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``GMLDataSet``.
+
+        Args:
+            filepath: Filepath in POSIX format to the NetworkX GML file.
+            load_args: Arguments passed on to ```networkx.read_gml``.
+                See the details in
+                https://networkx.org/documentation/stable/reference/readwrite/generated/networkx.readwrite.gml.read_gml.html
+            save_args: Arguments passed on to ```networkx.write_gml``.
```
Comment (AntonyMilneQB) [inline kedro/extras/datasets/networkx/gml_dataset.py:89]:
```suggestion
            load_args: Arguments passed on to ``networkx.read_gml``.
                See the details in
                https://networkx.org/documentation/stable/reference/readwrite/generated/networkx.readwrite.gml.read_gml.html
            save_args: Arguments passed on to ``networkx.write_gml``.
```

File: kedro/extras/datasets/networkx/graphml_dataset.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        fs_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``GraphMLDataSet``.
+
+        Args:
+            filepath: Filepath in POSIX format to the NetworkX GraphML file.
+            load_args: Arguments passed on to ```networkx.read_graphml``.
+                See the details in
+                https://networkx.org/documentation/stable/reference/readwrite/generated/networkx.readwrite.graphml.read_graphml.html
+            save_args: Arguments passed on to ```networkx.write_graphml``.
```
Comment (AntonyMilneQB) [inline kedro/extras/datasets/networkx/graphml_dataset.py:88]:
```suggestion
            load_args: Arguments passed on to ``networkx.read_graphml``.
                See the details in
                https://networkx.org/documentation/stable/reference/readwrite/generated/networkx.readwrite.graphml.read_graphml.html
            save_args: Arguments passed on to ``networkx.write_graphml``.
```

**Review by AntonyMilneQB — APPROVED** [review]
This is awesome work, thank you very much @sbrugman for your contribution! Just a few minor comments, but I think this is a great improvement. Thank you!

File: docs/source/15_api_docs/kedro.extras.datasets.rst [hunk 5]
```diff
@@ -19,7 +19,9 @@ kedro.extras.datasets
    kedro.extras.datasets.holoviews.HoloviewsWriter
    kedro.extras.datasets.json.JSONDataSet
    kedro.extras.datasets.matplotlib.MatplotlibWriter
-   kedro.extras.datasets.networkx.NetworkXDataSet
+   kedro.extras.datasets.networkx.JSONDataSet
+   kedro.extras.datasets.networkx.GraphMLDataSet
+   kedro.extras.datasets.networkx.GMLDataSet
```
Comment (MerelTheisenQB) [inline docs/source/15_api_docs/kedro.extras.datasets.rst:24]: Very minor, but could you sort these alphabetically?

**Review by MerelTheisenQB — APPROVED** [review]
Looks good! 👍  Thanks for the contribution @sbrugman 😄

**Comment by sbrugman** [comment]
Thanks for the reviews @AntonyMilneQB and @MerelTheisenQB ! I've updated the branch to incorporate the suggestions.

**Review by idanov — APPROVED** [review]

**Pull request merged**